app/models/integration_collection_mappings/netsuite_collection_mapping.rb

```diff
@@ -3,6 +3,35 @@
 module IntegrationCollectionMappings
   class NetsuiteCollectionMapping < BaseCollectionMapping
     settings_accessors :tax_nexus, :tax_type, :tax_code
+    settings_accessors :currencies
+
+    validate :currency_mapping_format
+
+    private
+
+    def currency_mapping_format
+      # Other mapping_types shouldn't have currencies, but if they do, we validate the format
+      return if !currencies? && currencies.nil?
+
+      if currencies? && currencies.nil?
+        errors.add(:currencies, "value_is_mandatory")
+        return
+      end
+
+      if !currencies.is_a?(Hash)
+        errors.add(:currencies, "invalid_format")
+      elsif currencies.empty?
+        errors.add(:currencies, "cannot_be_empty")
+      elsif !currencies_hash_valid?
+        errors.add(:currencies, "invalid_format")
+      end
+    end
+
+    def currencies_hash_valid?
+      valid_currencies = Currencies::ACCEPTED_CURRENCIES.keys
+      currencies.keys.all? { it.is_a?(String) && valid_currencies.include?(it.to_sym) } &&
+        currencies.values.all? { it.is_a?(String) && it != "" }
```

Comment:
```suggestion
        currencies.values.all? { it.is_a?(String) && it.present? }
```

because:

```ruby
lago-api(dev)> "   ".present?
=> false
```